ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Main.java

```diff
@@ -146,15 +148,20 @@ public static void main(final String[] args) {
             LOG.info(CliCommand.getPlatformInformation());
             LOG.info(arguments.getArgumentLog());
 
+            // command help
             if (arguments.isHelp()) {
-                // no need to connect to vend help
-                final Ds3Cli runner = new Ds3Cli(null, arguments, null);
-                final CommandResponse response = runner.getCommandHelp();
-                System.out.println(response.getMessage());
-                System.exit(response.getReturnCode());
+                if (arguments.getHelp().equalsIgnoreCase("LIST_COMMANDS")) {
+                    System.out.println(CliCommandFactory.listAllCommands());
+                } else {
+                    System.out.println(CliCommand.getLongHelp(arguments.getHelp()));
+                    final CliCommand helpCommand = CliCommandFactory.getCommandExecutor(arguments.getHelp());
+                    helpCommand.printArgumentHelp(arguments);
+                }
+                System.exit(0);
             }
 
-            final Ds3Client client = createClient(arguments);
+
+            final Ds3Client client = arguments.createClient();
```

Comment:
Replace this with a function that takes the arguments as a parameter and then creates the client.